Fixes #20497: Add `range_contains` lookup and fix VLANGroup VID range filtering

pheus opened the pull request:
### Fixes: #20497

**Summary**
Introduce a generic ORM lookup `range_contains` for `ArrayField(RangeField)` to correctly match rows where a scalar value is contained by **any** range in the array (e.g., `VLANGroup.vid_ranges`). This replaces the raw‑SQL helper used by the VLAN group filter and aligns REST and GraphQL semantics with PostgreSQL range operators.

**Changes**
- **extras**: Add and register the `range_contains` lookup (guarded to `ArrayField(RangeField)`).
- **ipam/filtersets**: Refactor `VLANGroup` FilterSet (`contains_vid`) to use the lookup (remove raw SQL).
- **GraphQL schema**:
  - Add an `IntegerRangeArrayLookup` input and wire `VLANGroup.vid_ranges` to it.
  - Keep the public API friendly: clients use `vid_ranges: { contains: <Int> }`, which maps internally to `__range_contains`.

**Tests**
- Add ORM unit tests for the lookup (positive, negative, and boundary cases).
- Add REST FilterSet tests for `?contains_vid=<int>` (hit/miss and exclusive upper bound).

jeremystretch commented on netbox/extras/lookups.py:
```diff
@@ -1,9 +1,39 @@
+    Works with int4range[], int8range[], daterange[], tstzrange[], etc.
+    """
+
+    lookup_name = 'any_range_contains'
```
What do you think about shortening this a bit?

```suggestion
    lookup_name = 'range_contains'
```

jeremystretch commented on netbox/ipam/tests/test_filtersets.py:
```diff
@@ -1724,6 +1724,15 @@ def test_contains_vid(self):
         params = {'contains_vid': 1}
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 8)
 
+    def test_contains_vid_upper_bound_exclusive(self):
+        # 11 is NOT in [1,11)
+        params = {'contains_vid': 12}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
+
+    def test_contains_vid_miss(self):
+        params = {'contains_vid': 4095}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 0)
+
```
IMO we should wrap these into `test_contains_vid()` above as they all test the same filter.

jeremystretch reviewed (commented):
Awesome work @pheus! Just had a couple minor suggestions.

pheus committed:
feat(extras): Add range_contains ORM lookup

Introduce a generic lookup for ArrayField(RangeField) that matches rows
where a scalar value is contained by any range in the array
(e.g. VLANGroup.vid_ranges).
Replace the raw-SQL helper in the VLANGroup FilterSet (`contains_vid`)
with the ORM lookup for better maintainability.
Add tests for the lookup and the FilterSet behavior.

Closes #20497

jeremystretch reviewed (approved):
Thanks @pheus!
